File: backend/NordicWebHub.Api/Models/SeoReport.cs
namespace NordicWebHub.Api.Models;

public class SeoReport
{
    public int Id { get; set; }

    public int CompanyId { get; set; }

    public Company Company { get; set; } = null!;

    public int SeoScore { get; set; }

    public string TopKeywords { get; set; } = string.Empty;

    public string TechnicalIssues { get; set; } = string.Empty;

    public string Recommendations { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/nordicwebhub-client/src/types/seoReport.ts
export type SeoReport = {
  id: number
  companyId: number
  companyName: string
  seoScore: number
  topKeywords: string
  technicalIssues: string
  recommendations: string
  createdAt: string
}

export type CreateSeoReportDto = {
  companyId: number
  seoScore: number
  topKeywords: string
  technicalIssues: string
  recommendations: string
}

export type UpdateSeoReportDto = CreateSeoReportDto
